src/System.Diagnostics.EventLog/src/System/Diagnostics/EventLogEntry.cs

```diff
@@ -0,0 +1,484 @@
+                    string msg = owner.FormatMessageWrapper(dllNames, (uint)msgId, ReplacementStrings);
+                    if (msg == null)
+                    {
+                        StringBuilder msgBuf = new StringBuilder(SR.MessageNotFormatted + msgId + Source);
+                        string[] strings = ReplacementStrings;
+                        for (int i = 0; i < strings.Length; i++)
+                        {
+                            if (i != 0)
+                                msgBuf.Append(", ");
+                            msgBuf.Append("'");
+                            msgBuf.Append(strings[i]);
+                            msgBuf.Append("'");
+                        }
+
+                        msg = msgBuf.ToString();
+                    }
+                    else
+                        msg = ReplaceMessageParameters(msg, ReplacementStrings);
+                    
+                    message = msg;
+                }
+
+                return message;
+            }
+        }
+
+        [MonitoringDescription("The name of the application that wrote this entry.")]
+        public string Source
+        {
+            get
+            {
+                StringBuilder buf = new StringBuilder();
+                int pos = bufOffset + FieldOffsets.RAWDATA;
+
+                char ch = CharFrom(dataBuf, pos);
+                while (ch != '\0')
+                {
+                    buf.Append(ch);
+                    pos += 2;
+                    ch = CharFrom(dataBuf, pos);
+                }
+
+                return buf.ToString();
+            }
+        }
+
+        [MonitoringDescription("The application-supplied strings used in the message.")]
+        public string[] ReplacementStrings
+        {
+            get
+            {
+                string[] strings = new string[ShortFrom(dataBuf, bufOffset + FieldOffsets.NUMSTRINGS)];
+                int i = 0;
+                int bufpos = bufOffset + IntFrom(dataBuf, bufOffset + FieldOffsets.STRINGOFFSET);
+                StringBuilder buf = new StringBuilder();
+                while (i < strings.Length)
+                {
+                    char ch = CharFrom(dataBuf, bufpos);
+                    if (ch != '\0')
+                        buf.Append(ch);
+                    else
+                    {
+                        strings[i] = buf.ToString();
+                        i++;
+                        buf = new StringBuilder();
+                    }
+
+                    bufpos += 2;
+                }
+
+                return strings;
+            }
+        }
+
+        [
+            MonitoringDescription("The full number identifying the message in the event message dll."),
+            ComVisible(false)
+        ]
+        public Int64 InstanceId
+        {
+            get
+            {
+                return (UInt32)IntFrom(dataBuf, bufOffset + FieldOffsets.EVENTID);
+            }
+        }
+
+#if false
```

Comment:
what is that? #if false is always false. This code never gets compiled. If this is code for debugging I would put it under #if DEBUG

Reply:
not sure what the convention is here. cc @danmosemsft

Reply:
How to know if its debug or redundant

Reply:
delete it